ee/clickhouse/sql/person.py

```diff
@@ -160,6 +154,62 @@
     table_name=PERSONS_DISTINCT_ID_TABLE, cluster=CLICKHOUSE_CLUSTER, database=CLICKHOUSE_DATABASE,
 )
 
+#
+# person_distinct_ids2 - new table!
+#
+
+PERSON_DISTINCT_ID2_TABLE = "person_distinct_id2"
+
+PERSON_DISTINCT_ID2_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER {cluster}
+(
+    team_id Int64,
+    distinct_id VARCHAR,
+    person_id UUID,
+    is_deleted Boolean,
+    version Int64 DEFAULT 1
+    {extra_fields}
+) ENGINE = {engine}
+"""
+
+PERSON_DISTINCT_ID2_TABLE_SQL = lambda: (
+    PERSON_DISTINCT_ID2_TABLE_BASE_SQL
+    + """
+    ORDER BY (team_id, distinct_id)
+    SETTINGS index_granularity = 512
+    """
+).format(
+    table_name=PERSON_DISTINCT_ID2_TABLE,
+    cluster=CLICKHOUSE_CLUSTER,
+    engine=table_engine(PERSON_DISTINCT_ID2_TABLE, "version", REPLACING_MERGE_TREE, sharded=False),
+    extra_fields=KAFKA_COLUMNS,
+)
+
+KAFKA_PERSON_DISTINCT_ID2_TABLE_SQL = PERSON_DISTINCT_ID2_TABLE_BASE_SQL.format(
+    table_name="kafka_" + PERSON_DISTINCT_ID2_TABLE,
+    cluster=CLICKHOUSE_CLUSTER,
+    engine=kafka_engine(KAFKA_PERSON_DISTINCT_ID),
+    extra_fields="",
+)
+
+# You must include the database here because of a bug in clickhouse
+# related to https://github.com/ClickHouse/ClickHouse/issues/10471
+PERSON_DISTINCT_ID2_MV_SQL = """
+CREATE MATERIALIZED VIEW {table_name}_mv ON CLUSTER {cluster}
+TO {database}.{table_name}
+AS SELECT
+team_id,
+distinct_id,
+person_id,
+is_deleted,
+version,
+_timestamp,
+_offset
```

Comment:
I'm not sure how useful `_offset` is without `_partition`

Reply:
Good point, will add it.

Reply:
Added.